src/mapping/grouping/pairwise_comparison.jl

```diff
@@ -0,0 +1,116 @@
+export GroupViaPairwiseComparison
+
+"""
+    GroupViaPairwiseComparison(; distance_threshold::Real, kwargs...)
+
+Initialize a struct that contains instructions on how to group features in
+[`AttractorsViaFeaturizing`](@ref). `GroupViaPairwiseComparison` groups features and
+identifies clusters by considering the pairwise distance between features. It can be used
+as an alternative to the clustering method in `GroupViaClustering`, having the
+advantage that it is simpler, typically faster and uses less memory. 
+
+## Keyword arguments
+    
+* `distance_threshold`: a real number defining the maximum distance two features can be to
+  be considered in the same cluster - above the threshold, features are different. This
+  value simply needs to be large enough to differentiate clusters.
+* `distance_metric = Euclidean()`: A metric to be used in the clustering. It can be any
+  function `f(a, b)` that returns the distance between any type of data structure (usually
+  vectors or matrices of reals). Needs to be consistent with the `featurizer` function. All
+  metrics from Distances.jl can be used here.
```

Comment:
We can simplify here. First, let's do a mass-rename of `distance_metric` to simply `metric`. This is the name we use throughout the ecosystem. Then, let's do a mass rename of `distance_threshold` to just `threshold`, as similarly, that's the name we use. There is no reason to ambiguate with `distance` in front, because there is no other threshold in this functionality. Then, let's simply state as a description:

"A _function_ that given two features `metric(a, b)` it returns the distance between them. `a, b` are outputs of the `featurizer` function. Any `Metric` from Distances.jl may be used as well."